interpreter.go

```diff
@@ -415,21 +433,23 @@ func (i *interpreter) evaluate(a ast.Node) (value, error) {
 			bindEnv.upValues[bind.Variable] = th
 		}
 		i.newLocal(vars)
+		sz := len(i.stack.stack)
 		// Add new stack frame, with new thunk for this variable
 		// execute body WRT stack frame.
-		v, err := e.evalInCurrentContext(ast.Body)
-		i.stack.pop()
+		v, err := e.evalInCurrentContext(ast.Body, tc)
+		i.stack.pop(sz)
```

Comment:
why pop multiple things? can you add a comment

Reply:
This is not popping multiple frames. It pops one frame if it's the right frame, i.e. wasn't pushed before. I changed the name of the function to reflect that.